src/foam/lib/json/DateParser.java

```diff
@@ -12,7 +12,27 @@
   extends ProxyParser
 {
   public DateParser() {
-    super(new Alt(new LongParser(), 
+    super(new Alt(new LongParser(),
+                  new Seq1(1,
+                    new Literal("ISODate("),
```

Comment:
If this is only to accomodate existing data in MongoDB, I'd rather these parser here not be modified, and custom parser be added to a different package like foam.dao.mongo or something, or even just inside the MongoDAO.

Once the MongoDB data is migrated to a new FOAM DAO, these parsers won't be needed I believe.